### PR TITLE
Loopback framework

### PR DESCRIPTION
**Pull request by paoesco**
WYE backend is now using Loopback framework.

This first version has authentication module and retrieving rooms in mongoldb (rooms that are already present).

### CODE REVIEW

**Comment by romgar** [comment]
Do you mind creating another repository? I would rather keep my work on express easily accessible.

**Comment by paoesco** [comment]
Why not. I wanted to keep the repo clean. Shall I tag it ? Or maybe you could fork it ?
If you prefer the new repo, will do

**Comment by romgar** [comment]
Forked! Let's move on :-)

File: server/models/Room.json [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  },
+  "properties": {
+    "id": {
+      "type": "string",
+      "required": true
+    },
+    "Country": {
+      "type": "string"
+    },
+    "Region": {
```
Comment (romgar) [inline server/models/Room.json:20]: Do you think we really need this?

File: server/models/Room.json [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+      "type": "string"
+    },
+    "City": {
+      "type": "string"
+    },
+    "Organization": {
```
Comment (romgar) [inline server/models/Room.json]: Let's use "Organisation" to be british ;-) 

May be a more complex structure at some point.

File: server/models/Room.json [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+      "type": "string"
+    },
+    "Room": {
```
Comment (romgar) [inline server/models/Room.json]: Probably better to use `Name`, as `Room` is already the name of the model.

File: server/models/Room.json [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+      "type": "string"
+    },
+    "URL": {
+      "type": "string"
+    },
+    "Address": {
```
Comment (romgar) [inline server/models/Room.json:35]: We may have to split it at some point too.

**Review by romgar — COMMENTED** [review]
I trust the loopback init script! Small comments on the first model defined there.

File: server/models/Room.json (hunk 1 above)
Comment (paoesco) [reply server/models/Room.json:20]: Let's keep the info

File: server/models/Room.json (hunk 4 above)
Comment (paoesco) [reply server/models/Room.json:35]: Ok ! For now I KISS

**Pull request merged**